src/main/java/fr/esgi/al5/tayarim/services/LogementService.java

```diff
@@ -0,0 +1,291 @@
+   */
+  @Transactional
+  public LogementDto updateLogement(@NonNull Long id,
+      @NonNull LogementUpdateDto logementUpdateDto) {
+
+    if ((logementUpdateDto.getTitre() == null || logementUpdateDto.getTitre().isBlank())
+        && (logementUpdateDto.getNombresDeChambres() == null
+        || logementUpdateDto.getNombresDeChambres() == 0)
+        && (logementUpdateDto.getNombresDeLits() == null
+        || logementUpdateDto.getNombresDeLits() == 0)
+        && (logementUpdateDto.getNombresSallesDeBains() == null
+        || logementUpdateDto.getNombresSallesDeBains() == 0)
+        && (logementUpdateDto.getCapaciteMaxPersonne() == null
+        || logementUpdateDto.getCapaciteMaxPersonne() == 0)
+        && (logementUpdateDto.getNombresNuitsMin() == null
+        || logementUpdateDto.getNombresNuitsMin() == 0)
+        && (logementUpdateDto.getDescription() == null || logementUpdateDto.getDescription()
+        .isBlank())
+        && (logementUpdateDto.getPrixParNuit() == null || logementUpdateDto.getPrixParNuit() == 0)
+        && (logementUpdateDto.getDefaultCheckIn() == null || logementUpdateDto.getDefaultCheckIn()
+        .isBlank())
+        && (logementUpdateDto.getDefaultCheckOut() == null || logementUpdateDto.getDefaultCheckOut()
+        .isBlank())
+        && (logementUpdateDto.getIntervalReservation() == null
+        || logementUpdateDto.getIntervalReservation() == 0)
+        && (logementUpdateDto.getVille() == null || logementUpdateDto.getVille().isBlank())
+        && (logementUpdateDto.getAdresse() == null || logementUpdateDto.getAdresse().isBlank())
+        && (logementUpdateDto.getCodePostal() == null || logementUpdateDto.getCodePostal()
+        .isBlank())
+        && (logementUpdateDto.getPays() == null || logementUpdateDto.getPays().isBlank())
+        && (logementUpdateDto.getEtage() == null || logementUpdateDto.getEtage().isBlank())
+        && (logementUpdateDto.getNumeroDePorte() == null || logementUpdateDto.getNumeroDePorte()
+        .isBlank())
+        && (logementUpdateDto.getIdTypeLogement() == null
+        || logementUpdateDto.getIdTypeLogement() == 0)) {
+      throw new LogementInvalidUpdateBody();
+    }
+
+    Optional<Logement> optionalLogement = logementRepository.findById(id);
+
+    if (optionalLogement.isEmpty()) {
+      throw new LogementNotFoundException();
+    }
+
+    Logement logement = optionalLogement.get();
+
+    if (
+      (!logement.getIsLouable() && Boolean.parseBoolean(logementUpdateDto.getIsLouable()))
+      && (
+        (logement.getNombresDeChambres() == null && logementUpdateDto.getNombresDeChambres() == null)
+        || (logement.getNombresDeLits() == null && logementUpdateDto.getNombresDeLits() == null)
+        || (logement.getNombresSallesDeBains() == null && logementUpdateDto.getNombresSallesDeBains() == null)
+        || (logement.getCapaciteMaxPersonne() == null && logementUpdateDto.getCapaciteMaxPersonne() == null)
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 108).